pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -318,7 +321,7 @@ def on_test_batch_start(
         self.test_progress_bar.set_description(f"{self.test_description} DataLoader {dataloader_idx}")
 
     def on_test_batch_end(self, *_: Any) -> None:
-        if self._should_update(self.test_batch_idx, self.total_test_batches_current_dataloader):
```

Comment:
All these `self.total_*` properties are still used by Rich, correct?

Does that mean Rich also has this problem?

Reply:
ideally no.. since
```py
self.test_progress_bar.total == self.total_test_batches_current_dataloader
```
since I had to change it for the main progress bar, changed then here too.

But yes rich has some issues currently in some edge cases. Actually, I started with Rich, and ended up fixing tqdm.. there will be a follow-up for rich